### PR TITLE
Calculator 

### PR DESCRIPTION
**Pull request by Saqibi4213**
#Calculator Project

### Features Implemented:
- **Calculator Component**: Built a React-based calculator that meets all specified user stories.
- **Arithmetic Operations**: Implemented addition, subtraction, multiplication, and division with correct order of operations.
- **User Input Handling**: Ensured no leading zeros, limited to one decimal per number, and allowed continuous operations.
- **Display**: Dynamically updated display to reflect user input and calculation results.
- **Edge Case Handling**: Managed consecutive operators and negative numbers correctly.

### Tests and Validation:
- Passed all unit tests for required functionalities. and **FREE CODE CAMP**

### CODE REVIEW

**Comment by Saqibi4213** [comment]
### Greetings, dear Partners!

Hello @Nooria150 and @iamfarhatsharefi,

I would greatly appreciate it if you could take some time to review my repository and project. Your feedback is invaluable to me. Thank you!

**Comment by Saqibi4213** [comment]
**Hello Sir** @ahzamir

Due to some challenges my partners are facing, they are unable to provide feedback within the 24-hour timeframe. As a result, I am reaching out to you directly and submitting my project for your review.

I would greatly appreciate it if you could take a look at my work and provide your feedback at your earliest convenience.

Thank you for your understanding and support.

**Comment by iamfarhatsharefi** [comment]
@Saqibi4213 
well done my partner!
Created a professional Repo (You created a repository with your project name)
✔ Correct use of Gitflow (You created the development branch and made it your default branch, then created and committed your work on your feature branch)
✔ Commit messages are meaningful
✔ Setup the linters correctly (You added the correct linters config into your project according to the language you used on your code)
✔ No linter errors (All linters checks have been passed 🟢)
✔ You have followed the caalculator App project task and committed using the command line
✔ Created the pull request (From your feature branch to the development branch you created a pull request and submitted the pull request link for review)
✔ Good PR description (You added a meaningful description to your PR about your work and the features you added on this pull request)
✔Your code is clean
✔Your repo has a professional README.md file
✔ Your partner has reviewed your repo
✔Your design is responsive

**Comment by Nooria150** [comment]
hello @Saqibi4213
proud of you dear partner!
Created a professional Repo (You created a repository with your project name)
✔ Correct use of Gitflow (You created the development branch and made it your default branch, then created and committed your work on your feature branch)
✔ Commit messages are meaningful
✔ Setup the linters correctly (You added the correct linters config into your project according to the language you used on your code)
✔ No linter errors (All linters checks have been passed 🟢)
✔ You have followed the React clock App project task and committed using the command line
✔ Created the pull request (From your feature branch to the development branch you created a pull request and submitted the pull request link for review)
✔ Good PR description (You added a meaningful description to your PR about your work and the features you added on this pull request)
✔Your code is clean
✔Your repo has a professional README.md file
✔ Your partner has reviewed your repo
✔Your design is responsive

File: src/App.css
Comment (ahzamir) [inline src/App.css:1]: [OPTIONAL] -  Change these colors from black to white as it is not clear on this dark background

image

**Review by ahzamir — APPROVED** [review]
### Hi @Saqibi4213 🙋‍♀️,

> Excellent work! 💯 You have done a great job implementing all the project requirements. 🥳 Your feature is complete! Congratulations!!!🎊

- **[REQUIRED]** Share your project on this [GROUP](https://www.facebook.com/groups/kick-startdev), add a simple Proud Moment text and a screenshot of your project then besides that add your project link at the end of the text as a GitHub Source code reference and post.

- Share your Project on your LinkedIn account and add a proud moment text with a screenshot.

- Well Done great design Implemented. 🎯🎉👍
image

# Status: Approved ✅
There is nothing to say other than to merge your branch. 🎉🎯✅👍😊🥳

WinkThumbsUpGIF


 ## Highlights 🌞

🎉 Great Design Implemented 
✔️ Created a professional Repo (You created a repository with your project name)
✔️ Correct use of Gitflow (You created the development branch and made it your default branch, then created and committed your work on your feature branch)
✔️ Commit messages are meaningful
✔️ Setup the linters correctly (You added the correct linters config into your project according to the language you used on your code)
✔️ No linter errors (All linters checks have been passed 🟢)
✔️ You have followed the Javascript Calculator project task and committed using the command line
✔️ Created the pull request (From your feature branch to the development branch you created a pull request and submitted the pull request link for review)
✔️ Good PR description (You added a meaningful description to your PR about your work and the features you added on this pull request)
✔️Your code is clean
✔️Your repo has a professional README.md file
✔️ Your partner has reviewed your repo
✔️Your design is responsive

**Cheers 🥂 and Happy coding!!! 👯 Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Remember to tag me @ahzamir in your question so I can receive the notification.**

**Pull request merged**